File: frontend/src/types/quota.ts
export interface QuotaResource {
  resource: string;
  hard: string;
  used: string;
}

export interface ResourceQuotaInfo {
  name: string;
  namespace: string;
  resources: QuotaResource[];
  created_at: string | null;
}

export interface LimitRangeItem {
  type: string;
  resource: string;
  max: string | null;
  min: string | null;
  default: string | null;
  default_request: string | null;
}

export interface LimitRangeInfo {
  name: string;
  namespace: string;
  limits: LimitRangeItem[];
  created_at: string | null;
}

File: backend/app/schemas/quota.py
from pydantic import BaseModel


class QuotaResource(BaseModel):
    resource: str
    hard: str
    used: str


class ResourceQuotaInfo(BaseModel):
    name: str
    namespace: str
    resources: list[QuotaResource]
    created_at: str | None


class LimitRangeItem(BaseModel):
    type: str
    resource: str
    max: str | None
    min: str | None
    default: str | None
    default_request: str | None


class LimitRangeInfo(BaseModel):
    name: str
    namespace: str
    limits: list[LimitRangeItem]
    created_at: str | None
